app/(shared)/public/conversations/[conversationId]/messages/route.ts

```diff
@@ -0,0 +1,29 @@
+import { and, asc, eq } from "drizzle-orm";
+import { NextRequest, NextResponse } from "next/server";
+import { ZodError } from "zod";
+
+import { conversationMessagesResponseSchema } from "@/lib/api";
+import db from "@/lib/server/db";
+import * as schema from "@/lib/server/db/schema";
+
+export async function POST(request: NextRequest, { params }: { params: Promise<{ conversationId: string }> }) {
+  try {
+    const body = await request.json();
+    const { tenantId } = body;
+    const { conversationId } = await params;
+
+    // query directly in the route handler
+    const messages = await db
+      .select()
+      .from(schema.messages)
+      .where(and(eq(schema.messages.tenantId, tenantId), eq(schema.messages.conversationId, conversationId)))
+      .orderBy(asc(schema.messages.createdAt));
```

Comment:
This looks like a security bug.  If you know the tenant ID and the conversationId then you can access any conversation even if you aren't logged in.